### PR TITLE
Add a test to ensure IOperation and patterns play nice together.

### PR DESCRIPTION
**Pull request by gafter**
@jaredpar You asked to see this.

### CODE REVIEW

**Comment by gafter** [comment]
@dotnet/roslyn-analysis Can I please have a review of this unit test?

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs [hunk 1]
```diff
@@ -12,6 +12,11 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class OperationAnalyzerTests : CompilingTestBase
     {
+        private static CSharpParseOptions patternParseOptions =
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: `s_patternParseOptions`

Can this be readonly?

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs [hunk 2]
```diff
@@ -12,6 +12,11 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
...
+            TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp6)
+                    .WithFeature("patterns", "true")
+                    .WithFeature("patternsExperimental", "true");
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Can we grab these names via the `MessageID` structure?  Makes it more resilient to change.

**Comment by jaredpar** [comment]
:+1: modulo comments

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs (hunk 2 above)
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Yes, that would be a great idea for a global refactoring of the way we handle features. I'd prefer not to mix that with these changes.

Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Are you suggesting something like `Microsoft.CodeAnalysis.CSharp.MessageID.IDS_FeaturePatternMatching2.RequiredFeature()` etc?

Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Yes.  That's how I've been using the name in my other IDE tests.

Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: `Microsoft.CodeAnalysis.CSharp.MessageID` is internal, so not available in the IDE. I'll look to see where else those strings can be found.

**Pull request merged**